app/services/paper_converters/figure_proxy.rb

```
@@ -46,17 +46,24 @@ def self.from_figure(figure)
 
     def self.from_snapshot(snapshot)
       token = snapshot.get_property("url").split('/').last
-      resource_token = ResourceToken.find_by!(token: token)
+      resource_token = ResourceToken.find_by(token: token)
       new(
         title: snapshot.get_property("title"),
-        href: resource_token.url(:detail)
+        href: resource_token.try(:url, :detail),
```

should this fail hard?

---

fixed.